lib/matplotlib/backend_bases.py

```diff
@@ -2959,10 +3010,52 @@ def _mouse_event_to_message(event):
                 return s
         return ""
 
+    def _nonrect(self, x):
+        from .patches import Rectangle
+        return not isinstance(x, Rectangle)
+
+    def _tooltip_list(self, event, hover):
+        import matplotlib.pyplot as plt
+        lines = plt.gca().get_lines()
```

Comment:
This is rather suspicious... there is no requirement that the current pyplot axes are the axes for this event, etc...

In general pyplot should be avoided in all internal code.